pkg/go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/pulumi/pulumi/pkg/v3
 
-go 1.20
+go 1.21
```

Comment:
I think this needs to stay on 1.20. Providers still link to /pkg and technically we support 1.20 for provider authoring still.